source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -346,69 +375,83 @@ IR::FunctionType inferEnvoyFunctionType(R (*)(void*, Args...)) {
 using namespace Wavm;
 
 template <typename R, typename... Args>
-void registerCallbackWavm(WasmVm* vm, absl::string_view functionName, R (*f)(Args...)) {
+void registerCallbackWavm(WasmVm* vm, absl::string_view moduleName, absl::string_view functionName,
+                          R (*f)(Args...)) {
   auto wavm = static_cast<Common::Wasm::Wavm::Wavm*>(vm);
-  wavm->envoy_functions_.emplace_back(
-      new Intrinsics::Function(wavm->envoy_module_, functionName.data(), reinterpret_cast<void*>(f),
-                               inferEnvoyFunctionType(f), IR::CallingConvention::intrinsic));
+  wavm->envoyFunctions_.emplace_back(new Intrinsics::Function(
```

Comment:
```suggestion
  wavm->envoyFunctions_.emplace_back(std::make_unique<Intrinsics::Function>(
```

Reply:
In this case I don't see how that is an improvement. It is clear that the pointer is not lost, and in the second case (your suggestion) it provides the additional bit of information that the container is of unique_ptr but that bit of information is not salient useful at this point.

Reply:
Agree that at this moment the pointer is not lost. It reduces the risk when we make further changes.
Meanwhile my suggestion is self explaining: no extra thinking for any one doesn't have the full knowledge of `envoyFunctions_`